src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -75,23 +241,165 @@ export const CollectionNfts = ({ contractAddress }: CollectionNftsProps) => {
     }
   )
 
+  const syncLocalFiltersWithURL = useCallback((state: CollectionFilters) => {
+    const urlFilterItems = [
+      'markets',
+      'maxPrice',
+      'maxRarity',
+      'minPrice',
+      'minRarity',
+      'traits',
+      'all',
+      'search',
+      'sort',
+    ] as const
+
+    const query: Record<string, any> = {}
+    urlFilterItems.forEach((key) => {
+      switch (key) {
+        case 'traits':
+          const traits = state.traits.map(({ trait_type, trait_value }) => `("${trait_type}","${trait_value}")`)
+
+          query['traits'] = traits
+          break
+
+        case 'all':
+          query['all'] = !state.buyNow
+          break
+
+        case 'sort':
+          query['sort'] = SortByPointers[state.sortBy]
+          break
+
+        default:
+          query[key] = state[key]
+          break
+      }
+    })
+
+    const modifiedQuery = urlParamsUtils.removeDefaults(query)
+
+    // Applying local state changes to URL
+    const url = window.location.href.split('?')[0]
+    const stringifiedQuery = qs.stringify(modifiedQuery, { arrayFormat: 'comma' })
+
+    // Using pushState on purpose here. router.push() will trigger re-renders & API calls.
+    window.history.pushState({}, ``, `${url}${stringifiedQuery && `?${stringifiedQuery}`}`)
+  }, [])
+
+  const location = useLocation()
+  const applyFiltersFromURL = useCallback(() => {
+    if (!location.search) return
+
+    const query = qs.parse(location.search, {
+      arrayFormat: 'comma',
+      parseNumbers: true,
+      parseBooleans: true,
+    }) as {
+      maxPrice: string
+      maxRarity: string
+      minPrice: string
+      minRarity: string
+      search: string
+      sort: string
+      sortBy: number
+      all: boolean
+      buyNow: boolean
+      traits: string[]
+      markets: string[]
+    }
+    const modifiedQuery = urlParamsUtils.buildQuery(query, collectionStats)
+
+    requestAnimationFrame(() => {
```

Comment:
this is weird - the `useCollectionFilters.setState(modifiedQuery as any)` should probably happen in the `useEffect` where this function is called and `modifiedQuery` should be returned from this function